src/App.tsx

```diff
@@ -1,17 +1,171 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import { getTodos, deleteTodo, updateTodoStatus } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodosList } from './components/TodosList';
+import { FilterBy } from './types/FilterBy';
+import { FilterBar } from './components/FilterBar';
+import { TodoAdd } from './components/TodoAdd';
+import { Errors } from './components/Errors';
+import { ErrorsType } from './types/ErrorsType';
+
+import './styles/transition.css';
+
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  const [todosList, setTodoList] = useState<Todo[]>([]);
+
+  // Errors
+  const [errors, setErrors] = useState<ErrorsType[]>([]);
+
+  // states
+  const [isLoadingTodos, setIsLoadingTodos] = useState<number[]>([]);
+
+  // Filter By
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+
+  const leftTodos = useMemo(() => (
+    todosList.filter(todo => todo.completed === false).length
+  ), [todosList]);
+
+  const isSelectAll = leftTodos === 0;
+
+  const completedTodos = todosList.length - leftTodos;
+
+  const filteredTodos = useMemo(() => {
+    return todosList.filter(todo => {
+      switch (filterBy) {
+        case FilterBy.Active:
+          return todo.completed === false;
+        case FilterBy.Completed:
+          return todo.completed === true;
+        case FilterBy.All:
+        default:
+          return true;
+      }
+    });
+  }, [filterBy, todosList]);
+
+  const getTodosList = useCallback(async () => {
+    if (user) {
+      const response = await getTodos(user.id)
+        .catch(() => {
+          setErrors(currErrors => [
+            ...currErrors,
+            ErrorsType.Get,
+          ]);
+          setTimeout(() => {
+            setErrors(currErrors => currErrors
+              .filter(error => error !== ErrorsType.Get));
+          }, 3000);
+          throw new Error('Unable to get Todos from server');
+        });
+
+      setTodoList(response);
+    } else {
+      throw new Error('No user');
+    }
```

Comment:
why not try/catch?

Reply:
it's not critical, but still it's strange that you've used different syntax in different places